### DISCUSSION TITLE
Reading and setting the time

### DISCUSSION THREAD

**Discussion by rwl7532** [jkotlinski/durexforth#532] [Q&A] [answered]
I push data to hours, minutes, seconds and 10th seconds.  The 6526 datasheet mentions "latched" until "The clock will not start again until after 
a write to the 1Oths of seconds register." 
I push data in BCD format to the four registers ($dd0b,a,9,8) but upon c@ retrieval from those four registers the data is static. Why?
Also I see some code referring to "include tod" Where is tod found? Looks handy.

**Comment by Whammo** [comment]
Ah, yes, that would be #335 "NMI TOD clock alarm interrupt"
 
 TOD stuff is a little tricky. I think I worked most of it out.

**Reply by rwl7532** [reply]
Is the 6526 timer part of getting the TOD registers set and unlatched?
And hoping to get file referenced by "include tod"

**Reply by Whammo** [reply]
No, the timer registers are not involved with TOD. 
 Also the reference to include TOD was to what code is referenced by #335
  Thank you for your interest in durexForth!

**Comment by Whammo** [comment]
If you are refering to:
  Hacking the 'ok' prompt for a clock #386
   It is quite outdated.
    Or, perhaps something else?

**Reply by rwl7532** [reply]
Not specifically. 
I did break the ask only one question rule!
First question was:
I push data in BCD format to the four registers ($dd0b,a,9,8) but upon c@ retrieval from those four registers the data is static. Why?

**Reply by Whammo** [reply]
I don't know, but if the code in #335 doesn't work for you, it's likely a hardware problem.
  The registers should be written in this order:  
  hour min sec 10th
  Also, make sure the alarm bit is not set.
  0 $dd0f c! \ tod bit

**Reply by Whammo** [reply]
you can read the registers in any order, but you must read 10ths after reading hours, or the clock won't un-latch.

**Reply by Whammo** [reply]
Interesting. My TOD stuff has broken!
 Ah, the march of progress.

**Comment by Whammo** [comment, marked as answer]
Programmers Reference Guide sez:
   A specific sequence of events must be followed for proper setting and
  reading of TOD. TOD is automatically stopped whenever a write to the
  Hours register occurs. The clock will not start again until after a write
  to the 10ths of seconds register. This assures TOD will always start at
  the desired time. Since a carry from one stage to the next can occur at
    any time with respect to a read operation, a latching function is
  included to keep all Time Of Day information constant during a read
  sequence. All four TOD registers latch on a read of Hours and remain
  latched until after a read of 10ths of seconds. The TOD clock continues
  to count when the output registers are latched. If only one register is
  to be read, there is no carry problem and the register can be read "on
  the fly," provided that any read of Hours is followed by a read of 10ths
  of seconds to disable the latching.

**Comment by rwl7532** [comment]
I got it working!  Thanks for your patience. 
: setall  dup >r set10s  setsec setminute sethr r> set10s ;
2 32 25 4 setall

**Comment by Whammo** [comment]
I have also repaired #335  when durexForth defaulted to ``decimal`` it broke quite a few bits.
